app/Http/Controllers/RedirectController.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ *    Copyright 2015-2017 ppy Pty. Ltd.
+ *
+ *    This file is part of osu!web. osu!web is distributed with the hope of
+ *    attracting more community contributions to the core ecosystem of osu!.
+ *
+ *    osu!web is free software: you can redistribute it and/or modify
+ *    it under the terms of the Affero GNU General Public License version 3
+ *    as published by the Free Software Foundation.
+ *
+ *    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+ *    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *    See the GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace App\Http\Controllers;
+
+class RedirectController extends Controller
+{
+    public function __invoke(...$args)
+    {
+        // Redirect routes should be named 'redirect.<target>'
+        return ujs_redirect(route(explode('redirect.', \Route::currentRouteName(), 2)[1], ...$args));
```

Comment:
this should just be `$args`. Or remove the `...$args` above and use `func_get_args()` instead.